pkg/controller/instance/instance_controller.go

```diff
@@ -113,18 +113,22 @@ func (r *Reconciler) SetupWithManager(
 		Complete(r)
 }
 
-// eventFilter ignores DeleteEvents for pipe-pods only (marked with task.PipePodAnnotation). This is due to an
-// inherent race that was described in detail in #1116 (https://github.com/kudobuilder/kudo/issues/1116)
-// tl;dr: pipe-task will delete the pipe pod at the end of the execution. this would normally trigger another
-// Instance reconciliation which might end up copying pipe files twice. we avoid this by explicitly ignoring
-// DeleteEvents for pipe-pods.
 func eventFilter() predicate.Funcs {
 	return predicate.Funcs{
 		CreateFunc: func(event.CreateEvent) bool { return true },
 		DeleteFunc: func(e event.DeleteEvent) bool {
+			// eventFilter ignores DeleteEvents for pipe-pods only (marked with task.PipePodAnnotation). This is due to an
+			// inherent race that was described in detail in #1116 (https://github.com/kudobuilder/kudo/issues/1116)
+			// tl;dr: pipe-task will delete the pipe pod at the end of the execution. this would normally trigger another
+			// Instance reconciliation which might end up copying pipe files twice. we avoid this by explicitly ignoring
+			// DeleteEvents for pipe-pods.
 			return !isForPipePod(e)
 		},
-		UpdateFunc:  func(event.UpdateEvent) bool { return true },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// by comparing the Meta.Generation counter for Instances we effectively ignore Instance.Status sub-resource changes
+			_, isInstance := e.ObjectNew.(*kudov1beta1.Instance)
+			return !isInstance || (e.MetaNew.GetGeneration() != e.MetaOld.GetGeneration())
```

Comment:
btw. we're also ignoring updates to annotations, labels etc. Is it ok? 🤔 probably yes, just wanted to put it out there :)

Reply:
hmm @ANeumann82 also raised a good point there 🤔

Reply:
Updates to annotations and labels affect the actual resource and therefore should increment the generation, so they shouldn't be ignored.